### PR TITLE
vdbench| Dynamicly calculate dataset

### PR DESCRIPTION
**Pull request by Avilir**
Making the calculation of the dataset automatically depending on storage size.

Adding test template file for CephFS.
Making the 'workloads' test that will run for both RBD and CephFS with 3
difference workloads :
   - Light - 15% of storage capacity
   - Medium - 35% of storage capacity
   - Heavy - 70% of storage capacity

According issue  #1776

### CODE REVIEW

File: tests/e2e/performance/test_vdbench_workload.py [hunk 1]
```diff
@@ -198,6 +190,44 @@ def test_vdbench_workload(
         sf_data = templating.load_yaml(template)
 
         target_results = template + 'Results'
+
+        log.info('Calculating Storage size....')
+        ceph_cluster = CephCluster()
+        total_capacity = ceph_cluster.get_ceph_capacity()
+        assert (total_capacity > constants.VDBENCH_MIN_CAPACITY), \
+            "Storage capacity is too low for performance testing"
+        log.info(f'The Total usable capacity is {total_capacity}')
```
Comment (AviadP) [inline tests/e2e/performance/test_vdbench_workload.py:199]: I would replace lines 199 with 197, so log will be presented before assertion

File: tests/e2e/performance/test_vdbench_workload.py [hunk 2]
```diff
@@ -198,6 +190,44 @@ def test_vdbench_workload(
...
+            capacity_per_pod = constants.VDBENCH_CAP_PER_POD
+            total_dirs = width ** depth
+            log.info(f'The total dirs in the tree {total_dirs}')
+            log.info(f'Going to run with {load} % of the capacity load.')
+            tested_capacity = round(total_capacity * 1024 * load / 100)
+            log.info(f'Tested capacity is {tested_capacity} MB')
+            servers = round(tested_capacity / capacity_per_pod)
+
+            """
+                To spread the application pods evenly on all workers or application nods and at least 2 app pods
```
Comment (AviadP) [inline tests/e2e/performance/test_vdbench_workload.py:214]: typo in word "nods" (word 12)

File: tests/e2e/performance/test_vdbench_workload.py (hunk 1 above)
Comment (Avilir) [reply tests/e2e/performance/test_vdbench_workload.py:199]: i only want the log if not asserted.

File: tests/e2e/performance/test_vdbench_workload.py (hunk 2 above)
Comment (Avilir) [reply tests/e2e/performance/test_vdbench_workload.py]: fixed

File: tests/e2e/performance/test_vdbench_workload.py [hunk 3]
```diff
@@ -135,31 +136,20 @@ class TestVDBenchWorkload(E2ETest):
     The total number of files is calculating by : files * (width ** depth)
     """
 
+    @pytest.mark.parametrize('template', [
+        pytest.param("VDBench-Basic.yaml", marks=pytest.mark.workloads()),
```
Comment (ebenahar) [inline tests/e2e/performance/test_vdbench_workload.py:140]: suggestion:
you can import 'workloads' in line #17:
```python
from ocs_ci.framework.testlib import E2ETest, workloads
```
and here do:
```python
pytest.param("VDBench-Basic-FS.yaml", marks=workloads)
```

File: tests/e2e/performance/test_vdbench_workload.py [hunk 4]
```diff
@@ -198,6 +190,44 @@ def test_vdbench_workload(
         sf_data = templating.load_yaml(template)
 
         target_results = template + 'Results'
+
+        log.info('Calculating Storage size....')
+        ceph_cluster = CephCluster()
+        total_capacity = ceph_cluster.get_ceph_capacity()
+        assert (total_capacity > constants.VDBENCH_MIN_CAPACITY), \
+            "Storage capacity is too low for performance testing"
```
Comment (ebenahar) [inline tests/e2e/performance/test_vdbench_workload.py:198]: Please change to:
```suggestion
        assert total_capacity > constants.VDBENCH_MIN_CAPACITY, (
            "Storage capacity is too low for performance testing"
        )
```

File: tests/e2e/performance/test_vdbench_workload.py [hunk 5]
```diff
@@ -198,6 +190,44 @@ def test_vdbench_workload(
...
+            capacity_per_pod = constants.VDBENCH_CAP_PER_POD
+            total_dirs = width ** depth
+            log.info(f'The total dirs in the tree {total_dirs}')
+            log.info(f'Going to run with {load} % of the capacity load.')
+            tested_capacity = round(total_capacity * 1024 * load / 100)
+            log.info(f'Tested capacity is {tested_capacity} MB')
+            servers = round(tested_capacity / capacity_per_pod)
+
+            """
+                To spread the application pods evenly on all workers or application nodes and at least 2 app pods
```
Comment (ebenahar) [inline tests/e2e/performance/test_vdbench_workload.py:214]: Text and quotation marks are not aligned

Comment (Avilir) [reply tests/e2e/performance/test_vdbench_workload.py]: fixed.

File: tests/e2e/performance/test_vdbench_workload.py (hunk 3 above)
Comment (ebenahar) [reply tests/e2e/performance/test_vdbench_workload.py:140]: please address

Comment (Avilir) [reply tests/e2e/performance/test_vdbench_workload.py]: I changed it.

**Comment by Avilir** [comment]
> PR validation [Job UNSTABLE](https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/1551//testReport) (some or all tests failed).

The tests pass but since the last 2 tests ran on 70% of storage capacity, in the end the storage status was in WARN state and not OK which failed it

**Pull request merged**